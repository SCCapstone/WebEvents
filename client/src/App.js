import React from "react";
import "./App.css";
import TitleBar from "./TitleBar";
import SheetJSApp from "./sheet";
import RequestServer from "./RequestServer";
import FileUpload from "./FileUpload";
import Options from "./Options";
import FileDownloader from "./FileDownload";

const App = () => {
  return (
    <div className="App">
          <TitleBar />
          <Options />
          <SheetJSApp />
          <RequestServer />
          <FileUpload />
          <FileDownloader />
    </div>
  );
};

export default App;
