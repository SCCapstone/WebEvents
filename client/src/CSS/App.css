*{
    padding: 0;
    margin: 0;
    list-style: none;
    text-decoration: none;
}
.App {
    height: 100%;
    width: 100%;
    display: block;
    margin: 0;
    padding: 0;
    overflow: hidden;
}

body {
    padding: 0;
    margin: 0; 
}

img {
    max-width: 100%;
    max-height: 100%;
    display: block; /* remove extra space below image */
}

/*
    MainBody is a container to hold all central 
    elements excluding the titlebar and footer

    TODO: 
        * Make it segmented and centralized.
*/

.main-body {
    display: block;
    float: right;
}

.inner-main-body {
    width: 1000px;
    height: 100%;
    display: block;
    margin: 0 auto;
    background-color: white;
    text-align: center;
}


/*
Retired old app.css
Lam Nguyen
2020-03-29

#MainBody {
    display: block;
    flex-direction: row;
    justify-content: space-evenly;
    height:80vh;
    background-color: silver;
}
#MainBody-Left {
    display: block;
    flex-direction: column;
    justify-content: flex-start;

    height: auto;
    width: 20vw;
    border: solid silver;
}
#MainBody-Center {
    display: block;
    flex-direction: column;
    justify-content: flex-start;
    background-color: silver;


    height: auto;
    width: 55vw;
    border: solid silver;
}

#MainBody-Right {
    display: block;
    flex-direction: column;
    justify-content: flex-start;

    height: auto;
    width: 25vw;
    border: solid silver;

}

#padded-container-center {
    margin: 3vh;
    margin-top: 5vh;

    height: 75vh;
    background-color: white;
    border: 3px grey solid;
}
#padded-container-left-type{
    margin: 5vh; 
    background-color: white;

    height: 20vh;
    width: 17vw;
    border: 3px grey solid;

    text-align: center;
}
#padded-container-left-option {
    margin: 5vh; 
    margin-top: 0vh;
    background-color: white;

    height: 55vh;
    width: 17vw;

    border: 3px grey solid;
    text-align: center;
}
#padded-container-right-instructions {
    margin: 5vh;
    margin-left: 0vh;
    background-color: white;

    height: 75vh;
    width: 22vw;

    border: 3px grey solid;
}
.column-padding {
    margin: 3vw; 
    height: 100%;
}
*/

/*.App {
  border: 5px solid maroon;
  text-align: center;
  background: maroon;
  color: black;
  margin: 0px;
  padding: 20px;
}
*/