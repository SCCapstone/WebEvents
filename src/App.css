/*.App {
  border: 5px solid maroon;
  text-align: center;
  background: maroon;
  color: black;
  margin: 0px;
  padding: 20px;
}
*/
