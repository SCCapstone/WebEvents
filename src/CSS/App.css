/*.App {
  border: 5px solid maroon;
  text-align: center;
  background: maroon;
  color: black;
  margin: 0px;
  padding: 20px;
}
*/

/*
    MainBody is a container to hold all central 
    elements excluding the titlebar and footer

    TODO: 
        * Make it segmented and centralized.
*/
#MainBody {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    height: 100vh;
    background-color: silver;
}
#MainBody-Left {
    display: flex;
}
#MainBody-Center {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    background-color: white;

    width: 100vh;
    height: 100vh;
}
#MainBody-Left {
    display: flex;
}
.column-padding {
    height: 100%;
}