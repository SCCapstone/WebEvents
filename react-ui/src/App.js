import React, { useCallback, useEffect, useState } from "react";
import "./App.css";
import CSVImport from "./CSVImport";
import TitleBar from "./TitleBar";
import SheetJSApp from "./sheet";
import RequestServer from "./RequestServer";
import FileUpload from "./FileUpload";
import Options from "./Options";

function App {
  const [message, setMessage] = useState(null);
  const [isFetching, setIsFetching] = useState(false);
  const [url, setUrl] = useState('/api');

  const fetchData = useCallback(() => {
    fetch(url)
      .then(response => {
        if (!response.ok) {
          throw new Error(`status ${response.status}`);
        }
        return response.json();
      })
      .then(json => {
        setMessage(json.message);
        setIsFetching(false);
      }).catch(e => {
        setMessage(`API call failed: ${e}`);
        setIsFetching(false);
      })
  }, [url]);

  useEffect(() => {
    setIsFetching(true);
    fetchData();
  }, [fetchData]);

  return (
    <div className="App">
      <TitleBar />
          <SheetJSApp />
          <RequestServer />
          <FileUpload />
          <Options />
    </div>
  );
};

export default App;
