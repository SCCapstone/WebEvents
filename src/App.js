import React from "react";
import "./App.css";
import CSVImport from "./CSVImport";
import TitleBar from "./TitleBar";
import Options from "./Options";

const App = () => {
  return (
    <div className="App">
        <TitleBar />
        <Options />
        <CSVImport />
        
    </div>
  );
};

export default App;
