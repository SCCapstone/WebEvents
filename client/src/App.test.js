import React from 'react';
import ReactDOM from 'react-dom';
import SheetJSApp from "./components/sheet.js";
import App from './App';
import { create } from "react-test-renderer";

//makes sure that the app renders without crashing
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);

})
//makes sure that the sheet component matches the snapshot
describe("sheets component", () => {
    test("Matches the snapshot", () => {
        const sheet = create(<SheetJSApp />);
        expect(sheet.toJSON()).toMatchSnapshot();
    });
});

;
