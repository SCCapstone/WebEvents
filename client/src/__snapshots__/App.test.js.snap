// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sheets component Matches the snapshot 1`] = `
<div
  class="sheetjs"
>
  <div
    onDragEnter={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <div
      className="col-xs-1"
    >
      <input
        accept=".xlsx,.xlsb,.xlsm,.xls,.xml,.csv,.txt,.ods,.fods,.uos,.sylk,.dif,.dbf,.prn,.qpw,.123,.wb*,.wq*,.html,.htm"
        className="form-control"
        onChange={[Function]}
        type="file"
      />
    </div>
    <br />
    <br />
  </div>
</div>
`;
